docs/best-practices/log-container-sidecarset.md

```diff
@@ -0,0 +1,285 @@
+
+## EFK + SidecarSet(FileBeat) Practice
+### Install EFK (ElasticSearch, Kibana)
+There is a lot of documentation in the community to install EFK, this article is mainly deployed by way of Helm, refer to [Elastic Helm Charts](https://github.com/elastic/helm-charts).
+First of all, K8S cluster needs *StorageClass* for ElasticSearch PVC, this article uses the already created *alibabacloud-cnfs-nas*, as follows:
+```shell
+helm-charts% kubectl get storageclass
+NAME                       PROVISIONER                       RECLAIMPOLICY   VOLUMEBINDINGMODE      ALLOWVOLUMEEXPANSION   AGE
+alibabacloud-cnfs-nas      nasplugin.csi.alibabacloud.com    Delete          Immediate              true                   6d2h
+```
+
+Helm deploy ElasticSearch, Kibana, as follows:
+```shell
+helm-charts% helm repo add elastic https://helm.elastic.co
+helm-charts% helm repo update
+
+## Install ElasticSearch，set storage-class is alibabacloud-cnfs-nas
+helm-charts% helm install elasticsearch elastic/elasticsearch --version 7.16.3 --set persistence.annotations."volume.beta.kubernetes.io/storage-class"=alibabacloud-cnfs-nas -n elastic-system
+
+## Install Kibana
+helm-charts% helm install kibana elastic/kibana --version 7.16.3 --set service.type=LoadBalancer -n elastic-system
+```
+
+### FileBeat SidecarSet CRD
+Create FileBeat configuration (the ConfigMap be created under business namespace), as follows:
+```yaml
+apiVersion: v1
+data:
+  filebeat.yml: |
+    filebeat.inputs:
+    - type: log
+      paths:
+      - /var/log/*
+    output.elasticsearch:
+      host: '${NODE_NAME}'
+      hosts: '${ELASTICSEARCH_HOSTS:elasticsearch-master.elastic-system:9200}'
+kind: ConfigMap
+metadata:
+  name: filebeat-config
+```
+
+FileBeat SidecarSet Configuration, as follows:
+```yaml
+apiVersion: apps.kruise.io/v1alpha1
+kind: SidecarSet
+metadata:
+  name: filebeat-sidecarset
+spec:
+  selector:
+    # Pod labels that need to be injected into the sidecar container
+    matchLabels:
+      kruise.io/inject-filebeat: "true"
+  # sidecarSet is effective for the whole cluster by default, you can specify the scope of the effect through the namespace field
+ #namespace: ns-xxx
+  containers:
+  - args:
+    - -e
+    - -E
+    - http.enabled=true
+    env:
+    - name: POD_NAMESPACE
+    valueFrom:
+      fieldRef:
+        apiVersion: v1
+          fieldPath: metadata.namespace
+    image: docker.elastic.co/beats/filebeat:7.16.2
+    livenessProbe:
+      exec:
+        command:
+        - sh
+        - -c
+        - |
+          #!/usr/bin/env bash -e
+          curl --fail 127.0.0.1:5066
+    name: filebeat
+    readinessProbe:
+      exec:
+        command:
+        - sh
+        - -c
+        - |
+          #!/usr/bin/env bash -e
+          filebeat test output
+    resources:
+      limits:
+        cpu: "1"
+        memory: 200Mi
+      requests:
+        cpu: 100m
```

Comment:
we can comment below the yaml that one can set sidecar request to 0 so as to avoid introduce extra resource requirement, and we should also explain the side effect of zero request(qos class changed to Burstable)